### PR TITLE
Add command to remove inactive remote accounts from the local database

### PR DESCRIPTION
**Pull request by evanphilip**
Splitting PR #21995 into two.

Alternative fix for https://github.com/mastodon/mastodon/issues/9567 : absence of method to remove avatar and header images, leading to excessive disk usage.

This implements a new sub-command for `tootctl account` called ` cull-inactive` to remove remote accounts that appear inactive from the local database.

This is inspired by the script in https://github.com/mastodon/mastodon/issues/9567#issuecomment-1320934381 by @dustinwilson . This fix was found effective (eg: https://github.com/mastodon/mastodon/issues/9567#issuecomment-1319138000 by @WesleyAC) to control excessive disc usage, which has becoming the primary point of issue https://github.com/mastodon/mastodon/issues/9567.

### CODE REVIEW

File: lib/mastodon/accounts_cli.rb [hunk 1]
```diff
@@ -344,6 +344,45 @@ def cull(*domains)
...
+      removed_accounts = Concurrent::Set[]
+      processed, culled = parallelize_with_progress(
+        Account.where({ last_webfingered_at: Time.zone.at(0)..time_ago,
+          updated_at: Time.zone.at(0)..time_ago }).left_outer_joins(:user).where(user: { id: nil })
+      ) do |account|
+        next if account.local?
+        next if Follow.where(account: account).or(Follow.where(target_account: account)).count.positive?
+        removed_accounts << account.url
+        DeleteAccountService.new.call(account, reserve_username: false, skip_side_effects: true) unless options[:dry_run]
+        1
```
Comment (connorshea) [inline lib/mastodon/accounts_cli.rb:373]: I'm not particularly familiar with the CLI DSL used by Mastodon, but why return 1 here?

File: lib/mastodon/accounts_cli.rb [hunk 2]
```diff
@@ -344,6 +344,45 @@ def cull(*domains)
...
+      defaults to 90 days.
+    DESC
+    def cull_inactive
+      time_ago        = options[:days].days.ago
+      dry_run         = options[:dry_run] ? ' (DRY RUN)' : ''
+
+      removed_accounts = Concurrent::Set[]
+      processed, culled = parallelize_with_progress(
+        Account.where({ last_webfingered_at: Time.zone.at(0)..time_ago,
+          updated_at: Time.zone.at(0)..time_ago }).left_outer_joins(:user).where(user: { id: nil })
```
Comment (connorshea) [inline lib/mastodon/accounts_cli.rb:367]: I can never remember how this should be written exactly, but have you tested this to make sure it works? The last bit should maybe be `left_outer_joins(:user).where(users: { id: nil })` (plural `users` in the `where`).

**Review by connorshea — APPROVED** [review]
Two questions, but generally looks good

File: lib/mastodon/accounts_cli.rb (hunk 1 above)
Comment (evanphilip) [reply lib/mastodon/accounts_cli.rb:373]: The 1 is cumulated over the concurrent workers, so we get the total number of culled accounts.

**Comment by evanphilip** [comment]
Moved to #22149 since I cannot get the tests to run.

**Comment by evanphilip** [comment]
@connorshea I am really sorry about this mess. I ended up deleting my fork while attempting to fix CircleCI, so I had to reopen the pull request. 
I am logged into CicleCI, still no luck triggering the tests.

**Pull request closed without merging**